src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_connection/manage_direct_query_data_connections_table.tsx

```diff
@@ -64,10 +69,51 @@ export const ManageDirectQueryDataConnectionsTable: React.FC<ManageDirectQueryDa
   savedObjects,
   uiSettings,
   featureFlagStatus,
+  application,
 }) => {
+  const [observabilityDashboardsExists, setObservabilityDashboardsExists] = useState(false);
+  const [showIntegrationsFlyout, setShowIntegrationsFlyout] = useState(false);
+  const [integrationsFlyout, setIntegrationsFlyout] = useState<React.JSX.Element | null>(null);
+
+  const checkIfObservabilityDashboardsPluginIsInstalled = () => {
+    fetch('/api/status', {
+      headers: {
+        'Content-Type': 'application/json',
+        'osd-xsrf': 'true',
+        'accept-language': 'en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7,zh-TW;q=0.6',
+        pragma: 'no-cache',
+        'sec-fetch-dest': 'empty',
+        'sec-fetch-mode': 'cors',
+        'sec-fetch-site': 'same-origin',
+      },
+      method: 'GET',
+      referrerPolicy: 'strict-origin-when-cross-origin',
+      mode: 'cors',
+      credentials: 'include',
+    })
+      .then(function (response) {
+        return response.json();
+      })
+      .then((data) => {
+        for (let i = 0; i < data.status.statuses.length; ++i) {
+          if (data.status.statuses[i].id.includes('plugin:observabilityDashboards')) {
+            setObservabilityDashboardsExists(true);
+          }
+        }
+      })
+      .catch((error) => {
+        notifications.toasts.addDanger(
+          'Error checking Dashboards Observability Plugin Installation status.'
+        );
+        // eslint-disable-next-line no-console
+        console.error(error);
+      });
+  };
+
   const [data, setData] = useState<DataConnection[]>([]);
   const [isModalVisible, setIsModalVisible] = useState(false);
   const [modalLayout, setModalLayout] = useState(<EuiOverlayMask />);
+  const [selectedConnection, setSelectedConnection] = useState<string | undefined>('');
```

Comment:
nit. thinking we could initialize it as undefined to indicate initial state or no selected connection found.

Reply:
fixed